pkg/new_storage/general.go

```diff
@@ -135,7 +147,7 @@ func (bd *BackupDestination) BackupList() ([]Backup, error) {
 		if err != nil {
 			result = append(result, Backup{
 				metadata.BackupMetadata{
-					BackupName: strings.Trim(o.Name(), "/"),
+					BackupName: metadataFilePath,
```

Comment:
looks like you broke 
`clickhouse-backup list remote`
for other storage types
you should set `backup_name` here
not `backup_name/metadata.json`

Looks like you previous fixes with delimiter and recursive was wrong
could you clarify why `Bucket.Get` from COS on `cos.go`  Walk with delimiter=/ for !recursive was not work properly?








functionality

Reply:
> looks like you broke `clickhouse-backup list remote` for other storage types you should set `backup_name` here not `backup_name/metadata.json`
> 
> Looks like you previous fixes with delimiter and recursive was wrong could you clarify why `Bucket.Get` from COS on `cos.go` Walk with delimiter=/ for !recursive was not work properly?
> 
> functionality

Yes